src/main/java/com/algolia/search/AnalyticsConfig.java

```diff
@@ -1,44 +1,41 @@
 package com.algolia.search;
 
 import com.algolia.search.models.common.CallType;
-import java.util.Collections;
-import java.util.EnumSet;
-import java.util.List;
+import java.util.*;
 import javax.annotation.Nonnull;
 
-@SuppressWarnings({"WeakerAccess", "unused"})
+@SuppressWarnings({"unused"})
 public final class AnalyticsConfig extends ConfigBase {
 
-  /**
-   * Creates an {@link AnalyticsConfig} with the default host
-   *
-   * @param applicationID The Application ID
-   * @param apiKey The API Key
-   */
-  public AnalyticsConfig(String applicationID, String apiKey) {
-    super(applicationID, apiKey);
+  public static class Builder extends ConfigBase.Builder<Builder> {
 
-    List<StatefulHost> hosts =
-        Collections.singletonList(
-            new StatefulHost("analytics.algolia.com", EnumSet.of(CallType.READ, CallType.WRITE)));
+    /**
+     * Creates an {@link AnalyticsConfig} with the default host
+     *
+     * @param applicationID The Application ID
+     * @param apiKey The API Key
+     */
+    public Builder(@Nonnull String applicationID, @Nonnull String apiKey) {
+      super(applicationID, apiKey);
```

Comment:
I suggest to adjust `ConfigBase`'s constructor to make it receive default hosts (rationale: it is not an optional parameter for ConfigBase):
```
super(applicationID, apiKey, Collections.singletonList(...))
```